### PR TITLE
Add OpenCensus metric bridge specification

### PR DESCRIPTION
**Pull request by dashpole**
Second attempt at https://github.com/open-telemetry/opentelemetry-specification/pull/2732

Part of https://github.com/open-telemetry/opentelemetry-specification/issues/1175

Related to https://github.com/open-telemetry/opentelemetry-specification/pull/2730

## Changes

Specify that OpenCensus metric bridges implement the MetricProducer interface, and describe non-trivial transformations involved.

This follows the path taken by the [java opencensus-shim](https://github.com/open-telemetry/opentelemetry-java/blob/main/opencensus-shim/src/main/java/io/opentelemetry/opencensusshim/metrics/OpenCensusMetricProducer.java#L27).

### CODE REVIEW

**Comment by dashpole** [comment]
@jsuereth @jack-berg @reyang

File: specification/compatibility/opencensus.md [hunk 1]
```diff
@@ -187,11 +187,52 @@ OpenCensus specifies the following [HTTP Attributes](https://github.com/census-i
...
+OpenTelemetry SDK. This enables the bridge to work with both push and pull
+metric exporters.
+
+#### Known Incompatibilities
+
+* OpenTelemetry does not support OpenCensus' GaugeHistogram type; these metrics
+  MUST be dropped when using the bridge.
+* OpenTelemetry does not currently support context-based attributes (tags).
+* The OpenCensus bridge produces counters and distributions with a cumulative
+  AggregationTemporality. Many OpenTelemetry exporters do not support cumulative
```
Comment (reyang) [inline specification/compatibility/opencensus.md:235]: "Many OpenTelemetry exporters do not support cumulative..." - all the current exporters support cumulative https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/metrics/sdk_exporters.

File: specification/compatibility/opencensus.md [hunk 2]
```diff
@@ -187,11 +187,52 @@ OpenCensus specifies the following [HTTP Attributes](https://github.com/census-i
...
+### Usage
+
+The shim can be passed as an option to an OpenTelemetry
+[MeterProvider](../metrics/sdk.md#meterprovider) when configuring the
+OpenTelemetry SDK. This enables the bridge to work with both push and pull
+metric exporters.
+
+#### Known Incompatibilities
+
+* OpenTelemetry does not support OpenCensus' GaugeHistogram type; these metrics
```
Comment (reyang) [inline specification/compatibility/opencensus.md:231]:
```suggestion
* OpenTelemetry does not currently support OpenCensus' GaugeHistogram type; these metrics
```

File: specification/compatibility/opencensus.md (hunk 1 above)
Comment (dashpole) [reply specification/compatibility/opencensus.md:235]: Actually, given the new language in the spec about configurability of aggregation temporality, we can drop this sentence.

File: specification/compatibility/opencensus.md [hunk 3]
```diff
@@ -187,11 +187,49 @@ OpenCensus specifies the following [HTTP Attributes](https://github.com/census-i
...
-> API and no SDK usage of OpenCensus will be compatible.
+OpenTelemetry will provide an OpenCensus-Metrics-Shim component which
+implements the OpenTelemetry [MetricProducer](../metrics/sdk.md#metricproducer)
+interface. When Produce() is invoked, the shim collects metrics from the
+OpenCensus global state, converts the metrics to an OpenTelemetry metrics
+batch, and returns.
+
+### Requirements
+
+* This component MUST be an optional dependency.
```
Comment (reyang) [inline specification/compatibility/opencensus.md:198]: nit: maybe remove the trailing `.` for consistency (with the other bullet points)

**Review by reyang — APPROVED** [review]

File: specification/compatibility/opencensus.md [hunk 4]
```diff
@@ -187,11 +187,49 @@ OpenCensus specifies the following [HTTP Attributes](https://github.com/census-i
...
+
+### Instrumentation Scope
+
+The shim MUST add an instrumentation scope name and version which identifies
+the shim.
+
+### Usage
+
+The shim can be passed as an option to an OpenTelemetry
+[MeterProvider](../metrics/sdk.md#meterprovider) when configuring the
```
Comment (jack-berg) [inline specification/compatibility/opencensus.md:225]: To the meter provider? Not to the MetricReader?

Comment (dashpole) [reply specification/compatibility/opencensus.md]: fixed

File: specification/compatibility/opencensus.md (hunk 3 above)
Comment (dashpole) [reply specification/compatibility/opencensus.md]: done

File: specification/compatibility/opencensus.md [hunk 5]
```diff
@@ -187,11 +187,49 @@ OpenCensus specifies the following [HTTP Attributes](https://github.com/census-i
...
+
+### Requirements
+
+* This component MUST be an optional dependency
+* MUST NOT require OpenCensus to depend on OpenTelemetry
+* MUST require few or no changes to OpenCensus
+* MUST be compatible with push and pull exporters
+* MUST support Gauges, Counters, Cumulative Histograms, and Summaries
+* are NOT REQUIRED to support Gauge Histograms
+* MUST support exemplars
```
[the remark this answers is not in the document]
Comment (dashpole) [reply specification/compatibility/opencensus.md]: updated

**Review by jmacd — APPROVED** [review]

**Pull request merged**